File: Java/프로그래머스/2/152996. 시소 짝꿍/시소 짝꿍.java
import java.util.*;

class Solution {
    public long solution(int[] weights) {
        long answer = 0;
        Arrays.sort(weights);
        Map<Double, Integer> map = new HashMap<>();
        for (int weight : weights) {
            Double a = (double) weight;
            Double b = (weight * 2) / 3.0;
            Double c = weight / 2.0;
            Double d = (weight * 3) / 4.0;

            if (map.containsKey(a)) {
                answer += map.get(a);
            }
            if (map.containsKey(b)) {
                answer += map.get(b);
            }
            if (map.containsKey(c)) {
                answer += map.get(c);
            }
            if (map.containsKey(d)) {
                answer += map.get(d);
            }

            map.put(a, map.getOrDefault(a, 0) + 1);
        }
        return answer;
    }
}

File: Python/프로그래머스/2/152996. 시소 짝꿍/시소 짝꿍.py
def solution(weights):
    answer = 0
    d = {}
    weights.sort()
    for weight in weights:
        aa = float(weight)
        bb = weight * 2 / 3
        cc = weight / 2
        dd = weight * 3 / 4

        if aa in d:
            answer += d[aa]
        if bb in d:
            answer += d[bb]
        if cc in d:
            answer += d[cc]
        if dd in d:
            answer += d[dd]

        d[aa] = d.get(aa, 0) + 1

    return answer